mysql-test/suite/innodb/t/foreign_sql_mode.test

```diff
@@ -0,0 +1,127 @@
+--source include/have_innodb.inc
+--source alter_sql_mode.inc
+call mtr.add_suppression("InnoDB: In ALTER TABLE .* has or is referenced in foreign key constraints which are not compatible with the new table definition.");
+
+let $combination=`select regexp_replace('$MTR_COMBINATIONS', 'innodb,\|,innodb', '')`;
+
+let $copy_algo=`select ((strcmp(substring_index('$combination', ",", 1), "COPY") = 0) or (strcmp(substring_index('$combination', ",", -1), "COPY") = 0))`;
+
+let $inplace_algo=`select ((strcmp(substring_index('$combination', ",", 1), "INPLACE") = 0) or (strcmp(substring_index('$combination', ",", -1), "INPLACE") = 0))`;
+
+let $algorithm=COPY;
+if ($inplace_algo)
+{
+  let $algorithm=INPLACE;
+}
+let $sql_mode = `SELECT @@SQL_MODE`;
+let $error_code = 0;
+if ($sql_mode == "STRICT_TRANS_TABLES") {
+ let $error_code = ER_FK_COLUMN_NOT_NULL;
+}
+
+--echo # modify child column NOT NULL on UPDATE CASCADE..parent column NULL
+CREATE TABLE t1(f1 INT, f2 INT, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT, FOREIGN KEY(f1) REFERENCES t1(f2) ON UPDATE CASCADE)ENGINE=InnoDB;
+replace_result ,ALGORITHM=COPY '' ,ALGORITHM=INPLACE '';
+--error $error_code
+eval ALTER TABLE t2 MODIFY COLUMN f1 INT NOT NULL,ALGORITHM=$algorithm;
+INSERT INTO t1 VALUES(1, 1);
+INSERT INTO t2 VALUES(1);
+
+let $dml_error_code = ER_ROW_IS_REFERENCED_2;
+if ($sql_mode == "STRICT_TRANS_TABLES")
+{
+  let $dml_error_code = 0;
+}
+
+--error $dml_error_code
+UPDATE t1 SET f2= NULL;
+DELETE FROM t2;
+SELECT * FROM t1;
+DROP TABLE t2, t1;
```

Comment:
I thought that we could add a second `UPDATE` after the `DELETE`, to demonstrate that it will always succeed.